application/src/main/java/uk/gov/hmcts/dm/config/azure/AzureStorageConfiguration.java

```diff
@@ -34,15 +34,19 @@ public CloudBlobClient cloudBlobClient() throws URISyntaxException, InvalidKeyEx
     }
 
     @Bean
-    CloudBlobContainer cloudBlobContainer() throws URISyntaxException, InvalidKeyException, StorageException {
+    public CloudBlobContainer cloudBlobContainer() throws URISyntaxException, InvalidKeyException, StorageException {
         final CloudBlobContainer container = cloudBlobClient().getContainerReference(this.containerReference);
-        if (isAzureBlobStoreEnabled() && !container.exists()) {
+        if (isAzureBlobStoreEnabled() && shouldCheckExistence() && !container.exists()) {
             // Current behaviour is that system will hang for a very long time
             throw new AppConfigurationException("Cloub Blob Container does not exist");
         }
         return container;
     }
 
+    protected boolean shouldCheckExistence() {
+        return true;
```

Comment:
???

Reply:
Introduced as a testability support for the class, and used later in fixing and issue in mocking of the class emerged after boot upgrade.